### PR TITLE
[FIX] Add NCIT namespace

### LINKED ISSUE #139: [BUG] `purl:` prefix for healthy controls (NCIT) doesn't currently have a corresponding URL
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Expected Behavior

All used prefixes in a `bagel pheno`-generated .jsonld (e.g., example_synthetic.jsonld) should expand properly to URLs in the graph.

### Current Behavior

Currently, the controlled term `purl:NCIT_C94342` is being used as the identifier for the healthy control `SubjectGroup`. This should expand to http://purl.obolibrary.org/obo/NCIT_C94342, but currently the namespaces specified in the `@context` do not include the NCIT PURL:
```json
  "@context": {
    "nb": "http://neurobagel.org/vocab/",
    "snomed": "http://purl.bioontology.org/ontology/SNOMEDCT/",
    "nidm": "http://purl.org/nidash/nidm#",
    "cogatlas": "https://www.cognitiveatlas.org/task/id/",
...
```

I think a few things need to be decided/done to address this.
- [x] a prefix and corresponding URL for the NCIT namespace need to be added to mappings.py
- [x] decide whether we want to stick to `purl` for the prefix, or (probably) go with something more descriptive like `ncit`
- [x] update context generation function
- [x] (maybe) once the namespace is added, write a test in the CLI to check that all used prefixes exist in the context & have a corresponding `http...` value
- [x] update example files
  - data dictionaries and jsonld files (context and values)

### PR DESCRIPTION
**Pull request by alyssadai**
Closes #139

Changes proposed in this pull request:

- New prefix and URL defined for the term for healthy controls
- Example JSON files updated with new TermURL
- New test added that all prefixes used in our reference data dictionary have an entry in the `@context`
  - employs a new helper function that gets all values for instances of a specific key in a dictionary

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [x] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: bagel/tests/test_utility.py [hunk 1]
```diff
@@ -17,6 +17,46 @@ def get_test_context():
     return putil.generate_context()
 
 
+@pytest.fixture
+def get_values_by_key():
```
Comment (surchs) [inline bagel/tests/test_utility.py:21]: neat

File: bagel/tests/test_utility.py [hunk 2]
```diff
@@ -17,6 +17,46 @@ def get_test_context():
...
+    get_test_context, get_values_by_key, load_test_json, test_data_upload_path
+):
+    """Test that all namespace prefixes used in a comprehensive data dictionary have a corresponding URL in the @context."""
+    data_dict = load_test_json(
+        test_data_upload_path / "example_synthetic.json"
+    )
+    prefixes = list(get_values_by_key(data_dict, "TermURL"))
+
+    for idx, _ in enumerate(prefixes):
+        prefixes[idx] = prefixes[idx].split(":")[0]
```
Comment (surchs) [inline bagel/tests/test_utility.py:50]: :cherries: but I'd find this cleaner

```suggestion
prefixes = list(map(lambda term: term.split(":")[0], 
                                  get_values_by_key(data_dict, "TermURL")))
```

**Review by surchs — APPROVED** [review]
Looks great @alyssadai :tada:. 

I do think we  need to also update the docs 

- here: https://neurobagel.org/dictionaries/#diagnosis
- and here: https://neurobagel.org/term_naming_standards/#currently-used-namespaces

To make this complete - so maybe don't close the issue quite yet and have those quick changes as a docs PR.

:cook:

**Comment by alyssadai** [comment]
good call on the docs updates @surchs!

**Pull request merged**